### DISCUSSION TITLE
Optimizing materials' properties at different frequency in a single ray tracing

### DISCUSSION THREAD

**Discussion by ZitongLan** [NVlabs/sionna#527] [General]
Hi, thanks for the great work Sionna team.

Currently I am working on the "[Learning of Material Properties via Gradient Descent](https://colab.research.google.com/github/NVlabs/diff-rt/blob/master/Learning_Materials.ipynb)" from official implementation. My sionna version is most updated one. 

I am customizing my own radio material, like
```py
RadioMaterial("common",
    relative_permittivity=1.0,
    conductivity=60,
    scattering_coefficient=0,
    xpd_coefficient=0)
```

My target is to simulate the environment (do ray tracing) at the different frequencies at the same time to combine different frequency response as a training target, i.e., **I need to combine simulation results from different frequencies to get a final result and optimize**. 

I do notice that there is a _frequency_update_callback_ method for the RadioMaterial. However, I don't understand how to use it properly to define a trainable radio materials at certen frequencies. And also I don't know how to do ray tracing for different frequencies at the same time (the path can be the same). It seems that the scene can only config at a certain frequency and the output path variable don't come with a dimension to support N_freq. Is it possible to use batchsize channel to complete this function?

![image](https://github.com/user-attachments/assets/8ab270aa-b024-42b3-ad58-c6eab0c8da92)

**Comment by jhoydis** [comment]
Hi @ZitongLan,

Sionna RT currently does not support simulations at different frequencies in parallel.

The only thing you can do is to compute paths at one frequency, then change the `scene.frequency` property and do another path computation. However, the material properties typically change with the frequency. This is what the callback is used for. You can have a look at the implementation of the [ITU materials](https://github.com/NVlabs/sionna/blob/0a4a22ec484d80ed038bf9b30240dfa4a3f90216/sionna/rt/itu_materials.py#L32).
The material that you have defined above does not depend on the frequency.

Hope that helps.

**Reply by ZitongLan** [reply]
Thanks, that helps!
